### ISSUE TITLE
Firefox Extension : Error while sending to JabRef

### ISSUE THREAD

**Issue by Carambarbe** [JabRef/JabRef-Browser-Extension#278]
This is about extension for Mozilla Firefox 85.0.2 (64 bits) on Windows 10.
The extension is freshly installed and I can't export anything from my browser to JabRef.

I get the error : 
"**Error while sending to JabRef.
Please see the browsers error console for details.**"

I get these two errors in the console:
- Error reading native manifest file C:\Programmes\JabRef\jabref-firefox.json: file is referenced in the registry but does not exist 
- ExtensionError: No such native application org.jabref.jabref

In the add-on setting, it says : "_Connection status : Attempt to postMessage on disconnected port_".

Thank you !

**Comment by tobiasdiez** [comment]
The first error says that the jabref-firefox.json cannot be found. Did you moved the installation of JabRef to a different folder? In this case, you need to change the registry entry `HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\org.jabref.jabref` to point to the correct file. See https://docs.jabref.org/collect/jabref-browser-extension#windows for more details; alternatively a reinstall of Jabref should fix it.

**Comment by Carambarbe** [comment]
My JabRef is by default installed in "_C:\Program Files\JabRef_".
I already tried to modify the registry without any success. I also tried to reinstall JabRef, it did not work either. 

It says that "C:\Programmes\JabRef\jabref-firefox.json does not exist" but when I type this file path in my file explorer it opens the _json_ file.

**Comment by tobiasdiez** [comment]
I guess you have a symbolic link from Programmes to Program Files, and firefox is not resolving this for some reason. What error do you get when you change the registry entry to `C:\Program Files\JabRef\jabref-firefox.json` ?

**Comment by Carambarbe** [comment]
I already tried both _Programmes_ and _Program Files_.

I moved the JabRef file directly in C:\, and changed the registry entry by typing `REG ADD "HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\org.jabref.jabref" /ve /d "C:\JabRef\jabref-firefox.json" /f` with admin rights. However, I still got the same error `C:\Programmes\JabRef\jabref-firefox.json does not exist`...

**Comment by tobiasdiez** [comment]
Running `REG ADD` doesn't change the registry, it only adds a new entry but doesn't overwrite an existing one. You can change the existing entry by running `regedit` and then manually navigating to "HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\org.jabref.jabref".

**Comment by Carambarbe** [comment]
I already tried to manually edit the registry with the registry editor (not from the console), so I don't think that's the point here. 
Whatever the path I specified in the registry, I always get the error _C:\Programmes\JabRef\jabref-firefox.json: file is referenced in the registry but does not exist_, even though I wrote a very different path in the registry entry.

**Comment by agirlando** [comment]
Same error also for Linux64 (Mageia7). Firefox 78.8.0esr, latest version of Jabref installed with the available *.rpm

**Comment by RashiqAzhan** [comment]
The problem still exists; it's present with all the browsers (all 3 variants of firefox, iridium, vivaldi).
I get the "Specified native messaging host not found." in the extensions even after having added using PowerShell. I checked with the Registry Editor to see if the command stuck and it did.

**Comment by tobiasdiez** [comment]
I'm sorry to hear that this problem is still not resolved. Sadly there are so many variables and the browsers don't really say what's the problem. According to https://developer.chrome.com/docs/apps/nativeMessaging/#native-messaging-debugging, your error message indicates that either the manifest json is not found, or that the path specified in the json file is not correct.

I suggest you start by going through the manual installation steps https://docs.jabref.org/collect/jabref-browser-extension#windows (although it sounds like you already did this). In step 6, please make sure that there is only one key in either HKEY_LOCAL_MACHINE or HKEY_CURRENT_USER. If this doesn't work a bit of experimentation might help in diagnosing the error:
- In regedit, rename the path to the json to something that doesn't exist. Does the error message change?
- In the json file, rename the path to the jabref executable. Does the error message change?
- Remove the json file. Does the error message change?

I'm sorry that I cannot really give a solution...

**Comment by RashiqAzhan** [comment]
@tobiasdiez I understand and thank you for taking the time to reply.

For anyone who is still running into this issue, here is how I solved it.

1. Uninstall Jabref completely (along with the registry modifications). This step is easier said than done and I used [BCU (Bulk Crap Uninstaller)](https://github.com/Klocman/Bulk-Crap-Uninstaller) to remove everything from all the residual files to all the registry entries.
2. (Optional) Uninstall the extensions in your browser as well.
3. (Optional) Run an [SFC scan](https://support.microsoft.com/en-us/topic/use-the-system-file-checker-tool-to-repair-missing-or-corrupted-system-files-79aa86cb-ca52-166a-92a3-966e85d4094e) and [DISM restore](https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/what-is-dism) to fix any underlying Windows issue that may be causing Jabref to bug out.
```
sfc /scannowDISM.exe /Online /Cleanup-Image /Restorehealth
```
```
DISM.exe /Online /Cleanup-Image /Restorehealth
```
Run these till they produce 0 errors or something of that eilk. They may have to be run multiple times between reboots to have the desired effects.
4. Reboot and install Jabref from a fresh download or using Chocolatey.
5. Disable and enable (or reinstall if you followed step 2) without doing a single step as shown in the [guide](https://docs.jabref.org/collect/jabref-browser-extension).

The solution worked across all 3 browsers in a Windows machine. I hope someone finds this useful.

TLDR: Don't follow the guide in the Jabref website.

**Comment by tobiasdiez** [comment]
I'm happy you found a solution.

For everyone running into the same issues, please try first if a simple JabRef uninstall + reinstall works before you proceed with the steps of @RashiqAzhan. Some of them (like SFC + DISM) are only recommended for users with experience of what these commands do.

**Comment by tobiasdiez** [comment]
This seems to be resolved. Thanks everyone for being so patient and sharing their workarounds.

The next version of the browser extension and of JabRef should also report more meaningful error messages now; in particular if powershell is not installed.
